diesel/src/query_builder/select_statement/dsl_impls.rs

```diff
@@ -70,6 +71,31 @@ where
     }
 }
 
+// FIXME: Should we disable select_by when `.group_by` has been called?
+impl<SE, F, S, D, W, O, LOf, G, LC, Selection> SelectByDsl<Selection>
+    for SelectStatement<F, S, D, W, O, LOf, G, LC>
+where
+    SE: Expression,
+    Selection: Selectable<Expression = SE>,
+    Self: SelectDsl<SE>,
+    <Self as SelectDsl<SE>>::Output: SelectByQuery<Expression = SE>,
```

Comment:
I'm not sure why the `SelectByQuery` bound here is required. Instead of that bound there should be some bounds along the lines of `SelectBy<Selection>: SelectableExpression<F>` to verify that the select clause is fine for the given from clause.

Reply:
It's just what SelectDsl write.